primitives/avn-common/src/event_types.rs

```diff
@@ -308,67 +350,66 @@ impl LiftedData {
+    pub fn from_erc_20_contract_transfer_bytes(
+        data: Option<Vec<u8>>,
+        topics: Vec<Vec<u8>>,
+    ) -> Result<Self, Error> {
         // Structure of input bytes:
         // data --> amount (32 bytes) (big endian)
         // all topics are 32 bytes long
         // topics[0] --> event signature (can be ignored)
-        // topics[1] --> currency contract address (first 12 bytes are 0 and should be ignored)
-        // topics[2] --> receiver t2 public key (32 bytes)
+        // topics[1] --> ethereum sender address
+        // topics[2] --> ethereum receiver address (bridge contract)
 
         if data.is_none() {
-            return Err(Error::LiftedToPredictionMarketEventMissingData)
+            return Err(Error::LiftedEventMissingData)
         }
         let data = data.expect("Already checked for errors");
 
         if data.len() != WORD_LENGTH {
-            return Err(Error::LiftedToPredictionMarketEventBadDataLength)
+            return Err(Error::LiftedEventBadDataLength)
         }
 
         if topics.len() != 3 {
-            return Err(Error::LiftedToPredictionMarketEventWrongTopicCount)
+            return Err(Error::LiftedEventWrongTopicCount)
         }
 
-        if topics[Self::TOPIC_CURRENCY_CONTRACT].len() != WORD_LENGTH ||
-            topics[Self::TOPIC_INDEX_T2_ADDRESS].len() != WORD_LENGTH
+        if topics[Self::TOPIC_BRIDGE_CONTRACT].len() != WORD_LENGTH ||
+            topics[Self::TOPIC_T1_FROM_ADDRESS].len() != WORD_LENGTH
         {
-            return Err(Error::LiftedToPredictionMarketEventBadTopicLength)
+            return Err(Error::LiftedEventBadTopicLength)
         }
 
-        let token_contract = H160::from_slice(
-            &topics[Self::TOPIC_CURRENCY_CONTRACT][DISCARDED_ZERO_BYTES..WORD_LENGTH],
+        let bridge_contract = H160::from_slice(
+            &topics[Self::TOPIC_BRIDGE_CONTRACT][DISCARDED_ZERO_BYTES..WORD_LENGTH],
         );
 
-        let receiver_address = H256::from_slice(&topics[Self::TOPIC_INDEX_T2_ADDRESS]);
+        let sender_address = H160::from_slice(
+            &topics[Self::TOPIC_T1_FROM_ADDRESS][DISCARDED_ZERO_BYTES..WORD_LENGTH],
+        );
 
         if data[0..HALF_WORD_LENGTH].iter().any(|byte| byte > &0) {
-            return Err(Error::LiftedToPredictionMarketEventDataOverflow)
+            return Err(Error::LiftedEventDataOverflow)
         }
 
         let amount = u128::from_be_bytes(
             data[HALF_WORD_LENGTH..WORD_LENGTH]
                 .try_into()
                 .expect("Slice is the correct size"),
         );
-        return Ok(LiftedToPredictionMarketData { token_contract, receiver_address, amount })
+        return Ok(LiftedData {
+            token_contract: H160::zero(),
+            sender_address,
+            // This must be overwritten after the lift reach consensus with the account configured
+            // on the pallet
+            receiver_address: bridge_contract.into(),
```

Comment:
This makes me a bit nervous because we risk locking the funds in an account without any private key. Can we "default" it to the treasury maybe?

Reply:
We could, but it's cleaner if this is done in runtime, by the processor that handles the event.
To have the treasury at the client level we will need to extend the runtime API which has a lot of overhead